ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoTransactionsSpec.scala

```diff
@@ -426,6 +431,73 @@ private[dao] trait JdbcLedgerDaoTransactionsSpec extends OptionValues with Insid
     }
   }
 
+  // Test case for #6698
+  it should "return empty source when offset range is from the future" in {
+    val commands: Vector[(Offset, LedgerEntry.Transaction)] = Vector.fill(3)(singleCreate)
+    val beginOffsetFromTheFuture = nextOffset();
+    val endOffsetFromTheFuture = nextOffset()
+
+    for {
+      _ <- commands.traverse(x => store(x))
+
+      result <- ledgerDao.transactionsReader
+        .getFlatTransactions(
+          beginOffsetFromTheFuture,
+          endOffsetFromTheFuture,
+          Map(alice -> Set.empty[Identifier]),
+          verbose = true)
+        .runWith(Sink.seq)
+
+    } yield {
+      extractAllTransactions(result) shouldBe empty
+    }
+  }
+
+  it should "return all transactions in the specified offset range when iterating with pageSize = 2" in {
+    val pageSize = 2
+
+    def offsetGap(): Vector[(Offset, LedgerEntry.Transaction)] = {
+      nextOffset()
+      Vector.empty[(Offset, LedgerEntry.Transaction)]
+    }
+
+    // the order of `nextOffset()` calls is important
+    val beginOffset = nextOffset()
+
+    val commands: Vector[(Offset, LedgerEntry.Transaction)] =
+      Vector(singleCreate) ++ offsetGap ++
+        Vector.fill(2)(singleCreate) ++ offsetGap ++
+        Vector.fill(3)(singleCreate) ++ offsetGap ++ offsetGap ++
+        Vector.fill(5)(singleCreate)
+
+    val endOffset = nextOffset()
+
+    commands.size shouldBe 11
+
+    for {
+      _ <- commands.traverse(x => store(x))
+
+      ledgerDao2 <- LoggingContext.newLoggingContext { implicit logCtx =>
+        daoOwner(pageSize).acquire()
+      }.asFuture
+
+      response <- ledgerDao2.transactionsReader
```

Comment:
```suggestion
    // Simulates a gap in the offsets assigned to events, as they
    // can be assigned to party allocation, package uploads and
    // configuration updates as well
    def offsetGap(): Vector[(Offset, LedgerEntry.Transaction)] = {
      nextOffset()
      Vector.empty[(Offset, LedgerEntry.Transaction)]
    }

    // the order of `nextOffset()` calls is important
    val beginOffset = nextOffset()

    val commands: Vector[(Offset, LedgerEntry.Transaction)] =
      Vector(singleCreate) ++ offsetGap ++
        Vector.fill(2)(singleCreate) ++ offsetGap ++
        Vector.fill(3)(singleCreate) ++ offsetGap ++ offsetGap ++
        Vector.fill(5)(singleCreate)

    val endOffset = nextOffset()

    commands.size shouldBe 11

    for {
      _ <- commands.traverse(x => store(x))

      ledgerDao <- LoggingContext.newLoggingContext { implicit logCtx =>
        // eventsPageSize = 2 makes sure that a page is delimited
        // by an offset that has is absent from the events table
        daoOwner(eventsPageSize = 2).acquire()
      }.asFuture

      response <- ledgerDao.transactionsReader
```
1. I don't see another `ledgerDao` in this scope, what does the `2` stand for?
2. `pageSize` is only used in a single place, I think that a named parameter makes it just as explicit without needing to go back to the beginning of the test for reference (or needing to keep track of bindings in your head)
3. I'd recommend adding a comment to make sure the meaning of `offsetGap` is clear
4. I'd recommend adding a comment to make sure that it's clear why `eventsPageSize = 2` makes this a good test case

Reply:
1. there is one ledgerDao in the scope, I wanted to show that this is not the default one.
adding comments all over test is a bad practice, they will get obsolete pretty soon.

Reply:
If you don't want to comment then please find some other way to make the following points clear:
1. why did we choose `eventPageSize` to be equal to 2
2. why are we putting gaps in the events: what do they represent and how are they meaningful

Otherwise the meaning of the test gets lost and the next person to read it may make wrong decisions with respects of the content of the test, possibly causing regressions.